vm/src/types/layout_name.rs

```diff
@@ -0,0 +1,68 @@
+#[cfg(all(feature = "arbitrary", feature = "std"))]
+use arbitrary::{self, Arbitrary};
+#[cfg(all(feature = "clap", feature = "std"))]
+use clap::{builder::PossibleValue, ValueEnum};
+use core::fmt::{self, Display};
+use serde::{Deserialize, Serialize};
+
+/// Enum representing the name of a Cairo Layout
+#[cfg_attr(all(feature = "arbitrary", feature = "std"), derive(Arbitrary))]
+#[derive(Serialize, Deserialize, Debug, PartialEq, Copy, Clone, Eq, Hash)]
+#[allow(non_camel_case_types)]
+pub enum LayoutName {
+    plain,
+    small,
+    dex,
+    recursive,
+    starknet,
+    starknet_with_keccak,
+    recursive_large_output,
+    all_solidity,
+    all_cairo,
+    dynamic,
+}
+
+impl LayoutName {
+    pub fn to_str(&self) -> &'static str {
```

Comment:
Since it's a fieldless enum, consider deriving `Copy` and making the methods take the value:
```suggestion
    pub fn to_str(self) -> &'static str {
```